src/utils/getArticleContents.js

```diff
@@ -0,0 +1,41 @@
+"use strict";
+
+const { parseHTML } = require("linkedom");
+
+/*
+ * Given page content markup, this function prepares a list of links to all headings of a given level
+ * followed by the page content with ID's injected for each linked heading.
+ *
+ * This work was undertaken to address GitHub issue #59:
+ * https://github.com/fluid-project/handbook.floeproject.org/issues/59
+ *
+ * @param {String} pageContent - the page content markup
+ * @param {String} [summary] - the summary text to display at the top of the article contents list
+ * @param {String} [headingsSelector] - a CSS selector to indicate which heading elements to list
+ * @param {String} [containerCssClass] - a CSS class for the article contents container element
+ *
+ * @return {String} - the combined string containing the list markup and modified page markup
+ */
+module.exports = (pageContent, summary = "Article Contents", headingsSelector = "h2", containerCssClass = "article-contents") => {
+    const {document} = parseHTML(pageContent);
+    const headings = document.querySelectorAll(headingsSelector);
+
+    const summaryCssClass = containerCssClass + "__summary";
+    const listCssClass = containerCssClass + "__list";
+    const listItemCssClass = containerCssClass + "__list-item";
+    const linkCssClass = containerCssClass + "__link";
+
+    var articleContentsList = "";
+
+    for (const heading of headings) {
```

Comment:
I'm a bit surprised this works because `heading` is being reassigned on each iteration. Perhaps it's because the nodes are objects? It should probably be a `let` instead of `const` here. Another option would be to call `headings.foreach((heading) => {...})`

Reply:
If I understand `const` correctly, even if it were an object, under other circumstances it would not allow us to redeclare or reassign to a declared `const`, but it seems that this value is redeclared on every iteration of the loop. `const` is block-scoped, so perhaps that's why it works. That said, I'm fairly certain this wouldn't be possible in a regular `for` loop, which is slightly confusing.

[MDN docs for a `for... of` loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of#syntax) say using `const`, `let` and `var` are all acceptable. I personally prefer using `const` since I feel it reinforces habits that prevent side effects by indicating that it's a value I shouldn't change, not only one that can't change.